include/pstl_offload/internal/usm_memory_replacement.h

```diff
@@ -18,7 +18,7 @@
 #include <atomic>
 #include <cassert>
 #include <cerrno>
-#include <optional>
+#include <mutex> // std::scoped_lock
```

Comment:
It seems like it is not needed anymore in this file

Reply:
Why? `std::scoped_lock` is used [below](https://github.com/oneapi-src/oneDPL/blob/902256bdb75f9005b4054f1d3f7d1d2ecd26c8db/include/pstl_offload/internal/usm_memory_replacement.h#L106).